src/libraries/System.Private.CoreLib/src/System/Random.LegacyImpl.cs

```diff
@@ -163,10 +163,9 @@ public override void NextBytes(byte[] buffer)
 
             public override void NextBytes(Span<byte> buffer)
             {
-                for (int i = 0; i < buffer.Length; i++)
-                {
-                    buffer[i] = (byte)_parent.Next();
-                }
+                byte[] tmpBuf = new byte[buffer.Length];
+                NextBytes(tmpBuf);
+                MemoryExtensions.CopyTo<byte>(tmpBuf, buffer);
```

Comment:
Thank you, but this isn't a change we can take.  Two reasons:
1. This was calling _parent.Next() because it was added after developers may have derived from Random and overridden the Next method.  If we didn't call Next, we would not end up using the developer's override.
2. This is allocating a new byte[] on every invocation.

Reply:
(At this point, as well, the "legacy" implementation is there just to handle the case of `new Random(seed)` or `new DerivedRandom()`, in which case we're more interested in compatibility than with throughput.)

Reply:
I see, that makes sense. I'll go ahead and close this PR then. Thanks for the info!